pkg/operation/hybridbotanist/machines.go

```diff
@@ -104,7 +104,7 @@ func (b *HybridBotanist) ReconcileMachines() error {
 
 	// Wait until all generated machine deployments are healthy/available.
 	if err := b.waitUntilMachineDeploymentsAvailable(wantedMachineDeployments); err != nil {
-		return common.DetermineErrorCode(fmt.Sprintf("Failed while waiting for all machine deployments to be ready: '%s'", err.Error()))
+		return common.NewTerraformError(fmt.Sprintf("Failed while waiting for all machine deployments to be ready: '%s'", err.Error()))
```

Comment:
Let's not call the function `NewTerraformError` as it is also used here where Terraform does not play any role at all.